h2/test_h2_headers.py

```diff
@@ -106,3 +107,184 @@ def test_random_header(self):
         self.start_tempesta()
         self.start_all_clients()
         self.wait_while_busy(curl)
+
+class CurlTestBase(tester.TempestaTest):
+
+    clients = [
+        {
+            'id' : 'curl',
+            'type' : 'external',
+            'binary' : 'curl',
+            'cmd_args' : (
+                '-kf ' # Set non-null return code on 4xx-5xx responses.
+                'https://${tempesta_ip}/ '
+                )
+        },
+    ]
+
+    def test(self, served_from_cache=False):
+        curl = self.get_client('curl')
+
+        self.start_all_servers()
+        self.start_tempesta()
+
+        self.start_all_clients()
+        self.wait_while_busy(curl)
+        self.assertEqual(0, curl.returncode,
+                         msg=("Curl return code is not 0 (%d)." %
+                              (curl.returncode)))
+        curl.stop()
+
+        self.start_all_clients()
+        self.wait_while_busy(curl)
+        self.assertEqual(0, curl.returncode,
+                         msg=("Curl return code is not 0 (%d)." %
+                              (curl.returncode)))
+
+        nginx = self.get_server('nginx')
+        nginx.get_stats()
+        self.assertEqual(1 if served_from_cache else 2, nginx.requests)
```

Comment:
Add a `msg` attribute here to provide a good explanation what happened wrong here. It's very handy when a test fails. Saves a lot of time on debugging.